### PR TITLE
 Misc improvements, mostly related to unit testing

### PR DESCRIPTION
**Pull request by Hilbrand**
- Rewrite MockChannel to use Mockito instead of implementing a mocked interface.
- Several sonar reported issues.
- Set threadnames for easier debugging.

### CODE REVIEW

File: source/taskmanager/src/main/java/nl/aerius/taskmanager/QueueWatchDog.java [hunk 1]
```diff
@@ -43,23 +42,21 @@
   public boolean isItDead(final boolean runningWorkers, final int numberOfMessages) {
     boolean doReset = false;
     if (runningWorkers && numberOfMessages == 0) {
-      if (firstProblem == 0) {
-        firstProblem = new Date().getTime();
+      if (firstProblem == null) {
+        firstProblem = now().plusMinutes(RESET_TIME_MINUTES);
```
Comment (SerhatG) [inline source/taskmanager/src/main/java/nl/aerius/taskmanager/QueueWatchDog.java:46]: I think it's still cleaner to do the check in the `else` like before.
Now we suggest that the first problem happened 10 minutes in the future.
Is there a reason this was moved to the `if`-statement?

File: source/taskmanager/src/main/java/nl/aerius/taskmanager/QueueWatchDog.java [hunk 2]
```diff
@@ -43,23 +42,21 @@
...
       }
     } else {
-      firstProblem = 0;
+      firstProblem = null;
     }
     return doReset;
   }
 
   /**
    * Convinces method for testing, because the time unit is in minutes, testing would take forever.
```
Comment (SerhatG) [inline source/taskmanager/src/main/java/nl/aerius/taskmanager/QueueWatchDog.java:57]: Do we want to update this? I am not exactly clear on what it says. Might just be a typo though.

File: source/taskmanager/src/main/java/nl/aerius/taskmanager/QueueWatchDog.java (hunk 1 above)
Comment (Hilbrand) [reply source/taskmanager/src/main/java/nl/aerius/taskmanager/QueueWatchDog.java]: It doesn't do the check here, but sets the time to measure to the time it should trigger. But it doesn't reflect the naming so I've moved it to the else part.

File: source/taskmanager/src/main/java/nl/aerius/taskmanager/QueueWatchDog.java (hunk 2 above)
Comment (Hilbrand) [reply source/taskmanager/src/main/java/nl/aerius/taskmanager/QueueWatchDog.java]: I've updated the comment. Hope it's clear now.

**Review by SerhatG — APPROVED** [review]

**Pull request merged**